test/verify/netlib.py

```diff
@@ -136,16 +136,18 @@ def iface_con_id(self, iface):
             return con_id
 
     def ensure_nm_uses_dhclient(self):
-        m = self.machine
-        m.write("/etc/NetworkManager/conf.d/99-dhcp.conf", "[main]\ndhcp=dhclient\n")
-        m.execute("systemctl restart NetworkManager")
+        self.write_file("/etc/NetworkManager/conf.d/99-dhcp.conf", "[main]\ndhcp=dhclient\n")
+        self.machine.execute("systemctl try-restart NetworkManager")
```

Comment:
We need to restart NM after restoring the file for non-destructive tests, right?

Reply:
That's right. This is done in PR #13879 in a more general way. So in that sense this change would be superfluous, but also harmless.